sdk/signalr/Microsoft.Azure.WebJobs.Extensions.SignalRService/api/Microsoft.Azure.WebJobs.Extensions.SignalRService.netstandard2.0.cs

```diff
@@ -104,23 +106,23 @@ public abstract partial class ServerlessHub : System.IDisposable
         protected virtual void Dispose(bool disposing) { }
         protected System.Collections.Generic.IList<System.Security.Claims.Claim> GetClaims(string jwt) { throw null; }
         protected Microsoft.Azure.WebJobs.Extensions.SignalRService.SignalRConnectionInfo Negotiate(string userId = null, System.Collections.Generic.IList<System.Security.Claims.Claim> claims = null, System.TimeSpan? lifetime = default(System.TimeSpan?)) { throw null; }
-        protected System.Threading.Tasks.Task<Microsoft.Azure.WebJobs.Extensions.SignalRService.SignalRConnectionInfo> NegotiateAsync(Microsoft.Azure.SignalR.Management.NegotiationOptions options) { throw null; }
+        protected virtual System.Threading.Tasks.Task<Microsoft.Azure.WebJobs.Extensions.SignalRService.SignalRConnectionInfo> NegotiateAsync(Microsoft.Azure.SignalR.Management.NegotiationOptions options) { throw null; }
```

Comment:
This could be a subtle breaking change, was this agreed on at arch board?

Reply:
>❌ DISALLOWED: Adding the virtual keyword to a member
While this often is not a breaking change because the C# compiler tends to emit callvirt Intermediate Language (IL) instructions to call non-virtual methods (callvirt performs a null check, while a normal call doesn't), this behavior is not invariable for several reasons:
C# is not the only language that .NET targets.
The C# compiler increasingly tries to optimize callvirt to a normal call whenever the target method is non-virtual and is probably not null (such as a method accessed through the ?. null propagation operator).
Making a method virtual means that the consumer code would often end up calling it non-virtually.

https://docs.microsoft.com/en-us/dotnet/core/compatibility/





Seems the only consequence is that when we call the virtual method from the base class, it might work as non-virtual. I think that's OK because currently calling many methods of  `ServerlessHub<T>` from a `ServerlessHub` is always not working now and users will get null pointer exception as we pass a null `ServiceHubContext` to the base class. `ServiceHubContext` cannot be converted from `ServiceHubContext<T>` yet and we might improve it later.